drivers/mtd_sdcard/mtd_sdcard.c

```diff
@@ -79,6 +63,28 @@ static int mtd_sdcard_read(mtd_dev_t *dev, void *buff, uint32_t addr,
     return -EIO;
 }
 
+static int mtd_sdcard_read_page(mtd_dev_t *dev, void *buff, uint32_t page,
+                                uint16_t offset, uint32_t size)
+{
+    DEBUG("mtd_sdcard_read_page: page:%" PRIu32 " offset:%" PRIu16 " size:%" PRIu32 "\n",
+          page, offset, size);
+
+    if (offset) {
+        return -ENOTSUP;
+    }
+
+    mtd_sdcard_t *mtd_sd = (mtd_sdcard_t*)dev;
+    sd_rw_response_t err;
+    sdcard_spi_read_blocks(mtd_sd->sd_card, page,
+                           buff, SD_HC_BLOCK_SIZE,
+                           size / SD_HC_BLOCK_SIZE, &err);
+
+    if (err == SD_RW_OK) {
+        return 0;
+    }
```

Comment:
What happens when `size % SD_HC_BLOCK_SIZE != 0`? The API doc does not contain a warning that the backend might require the size to a multiple of the block size.

How many MTD backends are expected to be support `offset != 0` and/or `size % BLOCK_SIZE != 0`? If there are few, maybe the API could drop the `offset` parameter and expect the number of blocks instead of the number of bytes?

Also: Is it only me, or is it super ugly that the SD-Card API returns the error via a pointer? Maybe this is something to touch in a follow up PR....

Reply:
All other backends support size != blocksize and offsets.
Currently the SD card backend only works with the FAT file system which will only ever operate on whole sectors.

I think it should be possible to modify `sdcard_spi` to support offsets on read - even if the SD card protocol only allows for sector-wise access, we can just ignore the first bytes.